File: src/types/tag.ts
/**
 * 技能词条（A/B/C 三级编号）
 */
export interface SkillEntry {
  /** 等级：A / B / C */
  level: string;
  /** 技能描述文本 */
  description: string;
}

/**
 * 标签数据模型
 */
export interface Tag {
  /** 标签唯一标识 */
  tag_id: string;
  /** 标签名称 */
  name: string;
  /** 技能词条列表 */
  skill_entries: SkillEntry[];
  /** 首次出现卡牌 */
  first_appearance: string;
  /** 设计初衷 */
  design_intent: string;
}

/**
 * 标记类型
 */
export interface Mark {
  /** 标记唯一标识 */
  mark_id: string;
  /** 标记名称 */
  name: string;
}

File: src-tauri/crates/core/src/tag.rs
// cardmaker-core: tag — 标签系统基础类型

use serde::{Deserialize, Serialize};

/// 标签
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Tag {
    pub id: String,
    pub name: String,
    pub color: String,
    pub description: String,
}

/// 标记（标签的实例，关联到具体卡牌）
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Mark {
    pub id: String,
    pub tag_id: String,
    pub card_id: String,
    pub note: String,
}